### PR TITLE
Add debug logging to metadata DB initialization

### PR DESCRIPTION
**Pull request by sondavidb**
**Issue #, if available:**

**Description of changes:**
Added a couple of extra debug-level logs at startup to better root-cause snapshotter startup issues.

**Testing performed:**

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cmd/soci-snapshotter-grpc/main.go [hunk 1]
```diff
@@ -289,9 +290,10 @@ const (
 	dbMetadataType = "db"
 )
 
-func getMetadataStore(rootDir string, config config.Config) (metadata.Store, error) {
+func getMetadataStore(ctx context.Context, rootDir string, config config.Config) (metadata.Store, error) {
 	switch config.MetadataStore {
 	case "", dbMetadataType:
+		log.G(ctx).Debug("initializing metadata db")
```
Comment (austinvazquez) [inline cmd/soci-snapshotter-grpc/main.go:296]: Maybe let's log `rootDir` as a field. It seems to be the only value that changes here. Everything else is static.

**Review by swagatbora90 — DISMISSED** [review]

File: cmd/soci-snapshotter-grpc/main.go [hunk 2]
```diff
@@ -289,9 +290,10 @@ const (
 	dbMetadataType = "db"
 )
 
-func getMetadataStore(rootDir string, config config.Config) (metadata.Store, error) {
+func getMetadataStore(ctx context.Context, rootDir string, config config.Config) (metadata.Store, error) {
 	switch config.MetadataStore {
 	case "", dbMetadataType:
+		log.G(ctx).Debugf("initializing metadata db at %v (type: %v)", rootDir, config.MetadataStore)
```
Comment (austinvazquez) [inline cmd/soci-snapshotter-grpc/main.go:296]: Let's use structured logging. See https://github.com/sirupsen/logrus?tab=readme-ov-file#fields

```suggestion
            log.G(ctx).WithField("root", rootDir).Debug("initializing metadata db")
```

**Comment by sondavidb** [comment]
Addresses @austinvazquez's comment above, as well as slightly modified logging messages in case we decide to use a different type of metadata storage in the future.

**Review by austinvazquez — APPROVED** [review]
+1, LGTM.

**Review by henry118 — APPROVED** [review]

**Pull request merged**